ts/client/services/syncer.ts

```diff
@@ -81,13 +139,47 @@ export class SyncerService {
         console.info(`#${batchID}  [${Usage[usage]}]`, batch.toString());
     }
 
+    getPubkeyFromTxn(txn: { data: string }): Pubkey {
+        // Get public key from registration call data
+        const { args } = this.accountRegistry.interface.parseTransaction(txn);
+        return new Pubkey(args.pubkey);
+    }
+
+    async handleSinglePubkeyRegistered(event: Event) {
+        const pubkeyID = event.args?.pubkeyID;
+
+        const txn = await event.getTransaction();
+        const pubkey = this.getPubkeyFromTxn(txn);
+
+        await this.pubkeyStorage.update(pubkeyID, pubkey);
+        await this.pubkeyStorage.commit();
+
+        console.info(`Pubkey added ID ${pubkeyID} ${pubkey.toString()}`);
+    }
+
+    async handleBatchPubkeyRegistered(_event: Event) {
+        console.error("handleBatchPubkeyRegistered not implemented");
+    }
+
     newBatchListener = async (
         batchID: null,
         accountRoot: null,
         batchType: null,
         event: Event
     ) => {
-        return await this.handleNewBatch(event);
+        return this.handleNewBatch(event);
```

Comment:
The function is async, why we don't await here?

Reply:
When chaining `async` functions, returning an `awaited` async function is mostly equivalent to just returning the async function.

![Screenshot from 2021-04-22 14-45-46](https://user-images.githubusercontent.com/10394161/115783128-b819c900-a379-11eb-9b5b-920e0bc67611.png)

This is also true when it throws/rejects, as it will bubble up to the top await.
That being said, all these functions return void values so we can just await rather than return. Will update.

See also this silliness:
![Screenshot from 2021-04-22 14-35-15](https://user-images.githubusercontent.com/10394161/115783374-062ecc80-a37a-11eb-9425-dde28dc00f86.png)

Reply:
Thanks for the explanation. That makes sense to me now.